src/qudi/hardware/adlink/fastcounter_adlink.py

```diff
@@ -0,0 +1,803 @@
+            "BufferAlloc": "Buffer allocation failed (WD_Buffer_Alloc)!",
+            "CH_Config": "Error while changing AI channel configuration (WD_AI_CH_Config)!",
+            "ContBufferSetup": "Error while setting continuous buffer (WD_AI_ContBufferSetup)!",
+            "ContScanChannels": "Error while starting continuous scan to file acquisition (WD_AI_ContScanChannels)!",
+            "AsyncClear": "Error while stopping continuous acquisition (WD_AI_AsyncClear)!",
+            "Register_Card": "Error while registering card (WD_Register_Card)!",
+            "GetDeviceProperties": "Error while getting device properties (WD_GetDeviceProperties)!",
+            "Config": "Error while setting AI config (WD_AI_Config)!",
+            "Trig_Config": "Error while setting AI trigger configuration (WD_AI_Trig_Config)!",
+            "SetLoggingDataCountPerFile": "Error while setting logging data count (WD_SetLoggingDataCountPerFile)!",
+            "SoftTriggerGen": "Error while sending a software trigger to the card (WD_SoftTriggerGen)!",
+            "AsyncDblBufferToFile": "Error while writing buffer to file. (WD_AI_AsyncDblBufferToFile)",
+            "AsyncReTrigNextReady": "Error while determining if next data is ready (WD_AI_AsyncReTrigNextReady)!",
+            "AsyncCheck": "Error while determining if next data is ready (WD_AI_AsyncCheck)!",
+            "EventCallBack": "Error when setting the callback function (WD_AI_EventCallBack_x64)!",
+            "AsyncDblBufferMode": "Error when setting double buffered mode (WD_AI_AsyncDblBufferMode)!",
+            "SetTimeout": "Error when setting acquisition timeout (WD_AI_SetTimeout)!",
+            "ContBufferReset": "Error when resetting Buffer (WD_AI_ContBufferReset)!",
+        }
+        self.log.error(
+            error_messages[error_str]
+            + f" ErrorCode {error_code.value}"
+            + " Reload module!"
+        )
+        try:
+            self.on_deactivate()
+        except Exception as e:
+            self.log.error(e)
+
+    def buffer_size_bytes(self):
+        """
+        Calculates the total size of the buffer by multiplying the byte size of the used data type
+        with the number of samples that are acquired.
+        """
+        buffer_size = self.buffer_size_samples().value * ctypes.sizeof(
+            self._settings.data_type
+        )
+        if buffer_size > self._maximum_samples * ctypes.sizeof(
+            self._settings.data_type
+        ):
+            buffer_size = self.buffer_size_samples().value * ctypes.sizeof(
+                self._settings.data_type
+            )
+
+        self._buffer_size_bytes = AdlinkDataTypes.U32(buffer_size)
+        return self._buffer_size_bytes
+
+    def buffer_size_samples(self):
+        """
+        Calculates the number of samples that will be acquired
+        """
+        buffer_size = self._settings.scancount_per_trigger.value * (
+            self._settings.channel_num.value + 1
```

Comment:
The card has the channel number setting which ranges from 0 to 3 for 4 channels. It can only activate chanels in sequence meaning channel_num = 1 means channel 1 and 2 are activated. In order to calculate the buffersize therefore the channel_num + 1 has to be taken.